tests/providers/google/common/hooks/test_base_google.py

```diff
@@ -632,6 +643,63 @@ def test_authorize_assert_http_timeout_is_present(self, mock_get_credentials):
         http_authorized = self.instance._authorize().http
         self.assertNotEqual(http_authorized.timeout, None)
 
+    @mock.patch(MODULE_NAME + '.get_credentials_and_project_id')
+    def test_get_credentials_and_project_id_with_impersonation_chain_as_string(
+        self,
+        mock_get_creds_and_proj_id
+    ):
+        mock_credentials = mock.MagicMock()
+        mock_get_creds_and_proj_id.return_value = (mock_credentials, "PROJECT_ID")
+        self.instance.impersonation_chain = "ACCOUNT_1"
```

Comment:
Does this create a global side effect? Or the instance is created in SetUp?

Reply:
`self.instance` is indeed created in `setUp`, so as far as I know, there should be no side effect.